pandas/tests/test_join.py

```diff
@@ -8,50 +9,46 @@
 
 
 class TestIndexer:
-    def test_outer_join_indexer(self):
-        typemap = [
-            ("int32", _join.outer_join_indexer_int32),
-            ("int64", _join.outer_join_indexer_int64),
-            ("float32", _join.outer_join_indexer_float32),
-            ("float64", _join.outer_join_indexer_float64),
-            ("object", _join.outer_join_indexer_object),
-        ]
-
-        for dtype, indexer in typemap:
-            left = np.arange(3, dtype=dtype)
-            right = np.arange(2, 5, dtype=dtype)
-            empty = np.array([], dtype=dtype)
-
-            result, lindexer, rindexer = indexer(left, right)
-            assert isinstance(result, np.ndarray)
-            assert isinstance(lindexer, np.ndarray)
-            assert isinstance(rindexer, np.ndarray)
-            tm.assert_numpy_array_equal(result, np.arange(5, dtype=dtype))
-            exp = np.array([0, 1, 2, -1, -1], dtype=np.int64)
-            tm.assert_numpy_array_equal(lindexer, exp)
-            exp = np.array([-1, -1, 0, 1, 2], dtype=np.int64)
-            tm.assert_numpy_array_equal(rindexer, exp)
-
-            result, lindexer, rindexer = indexer(empty, right)
-            tm.assert_numpy_array_equal(result, right)
-            exp = np.array([-1, -1, -1], dtype=np.int64)
-            tm.assert_numpy_array_equal(lindexer, exp)
-            exp = np.array([0, 1, 2], dtype=np.int64)
-            tm.assert_numpy_array_equal(rindexer, exp)
-
-            result, lindexer, rindexer = indexer(left, empty)
-            tm.assert_numpy_array_equal(result, left)
-            exp = np.array([0, 1, 2], dtype=np.int64)
-            tm.assert_numpy_array_equal(lindexer, exp)
-            exp = np.array([-1, -1, -1], dtype=np.int64)
-            tm.assert_numpy_array_equal(rindexer, exp)
+    @pytest.mark.parametrize(
+        "dtype", ["int32", "int64", "float32", "float64", "object"]
```

Comment:
at some point if can change to use a fixture to ensure more coverage (we may need to expand the indexers here as well), pls create a followup issue.